src/components/News.js

```diff
@@ -1,89 +1,104 @@
-import React, { useState, useEffect } from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import Newsitem from './Newsitem'
 import PropTypes from 'prop-types';
 import InfiniteScroll from 'react-infinite-scroll-component'
 import Loader from './Loader';
 
 const News = (props) => {
-
-    const [data, setData] = useState([]);
-    const [currentPage, setPage] = useState(1);
-    const [page, settotalPage] = useState(1)
-    const [loading, setLoading] = useState(true)
-
-    const capitalise = (str) => {
-        return (str.slice(0, 1)).toUpperCase() + str.slice(1)
+  
+  const [data, setData] = useState([]);
+  const [currentPage, setPage] = useState(1);
+  const [page, settotalPage] = useState(1)
+  const [loading, setLoading] = useState(true)
+
+  const capitalise = (str) => {
+    return (str.slice(0, 1)).toUpperCase() + str.slice(1)
+  }
+
+  const { setProgress } = props
+  const _setProgress = useCallback(
+    (val) => {
+      setProgress(val);
+    },
+    [setProgress]
+  );
+  
+  const updateNews = useCallback(async () => {
+    _setProgress(30);
+    setLoading(true);
+    const url = `https://newsapi.org/v2/top-headlines?q=${props.country}&apiKey=${props.apiKey}&page=1&pageSize=10&language=en&category=${props.category}`;
+    const data = await fetch(url);
```

Comment:
Added a bit of error handling for this fetch. Also below in the `fetchMoreData` function